hoot-core/src/main/cpp/hoot/core/schema/ScoreMatrix.h

```diff
@@ -45,18 +52,24 @@ namespace hoot
  * Hotel      | 0.02       | 0.9    | 0  | 0.2
  * California | 1.0        | 0.02   | 0  | 0.01
  */
+template<typename _type>
 class ScoreMatrix
 {
 public:
-  ScoreMatrix(int width, int height);
+  ScoreMatrix() { }
+  ScoreMatrix(int width, int height) { resize(width, height); }
+
+  void resize(size_t i, size_t j);
 
-  double get(size_t i, size_t j) const { return _v[i][j]; }
+  _type get(size_t i, size_t j) const { return _v[i][j]; }
+  _type& operator() (size_t i, size_t j) { return _v[i][j]; }
+  _type& operator() (size_t i, size_t j) const { return _v[i][j]; }
```

Comment:
Parenthesis operator for accessing the individual elements makes the code using the class look cleaner.